libursa/src/wasm/bls.rs

```diff
@@ -22,24 +22,27 @@ impl Generator {
 }
 
 #[wasm_bindgen]
+#[derive(Serialize, Deserialize)]
 pub struct SignKey(bls::SignKey);
 
 #[wasm_bindgen]
 impl SignKey {
-    pub fn new() -> Result<SignKey, JsValue> {
-        Ok(SignKey(maperr!(bls::SignKey::new(None))))
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> Result<JsValue, JsValue>  {
+        Ok(JsValue::from_serde(&SignKey(bls::SignKey::new(None)?)).unwrap())
     }
-
-    pub fn from_seed(seed: &[u8]) -> Result<SignKey, JsValue> {
-        Ok(SignKey(maperr!(bls::SignKey::new(Some(seed)))))
+    pub fn from_seed(seed: JsValue) -> Result<JsValue, JsValue> {
+        let _seed: Vec<u8> = seed.into_serde().unwrap();
```

Comment:
Please unwrap this correctly and don't use the unwrap function in production code

Reply:
I updated the code to use map_err instead of unwrap.